numpy/f2py/crackfortran.py

```diff
@@ -2481,11 +2504,31 @@ def get_parameters(vars, global_params={}):
                     # FIXME, unused l looks like potential bug
                     l = markoutercomma(v[1:-1]).split('@,@')
 
-            try:
-                params[n] = eval(v, g_params, params)
-            except Exception as msg:
-                params[n] = v
-                outmess('get_parameters: got "%s" on %s\n' % (msg, repr(v)))
+            # Identifying array parameters
+            param_is_array = False
+            for a in vars[n]['attrspec']:
+                if a[:9] == 'dimension':
+                    param_is_array = True
```

Comment:
nit: possibly clearer and certainly shorter as:
```suggestion
            param_is_array = any(a[:9] == 'dimension' for a in vars[n]['attrspec'])
```